contracts/modules/v2/Assessment.sol

```diff
@@ -0,0 +1,485 @@
+  ///
+  /// @param staker      The address of the staker for which the rewards are withdrawn
+  /// @param batchSize   The index until which (but not including) the rewards should be withdrawn.
+  ///                    Used if a large number of assessments accumulates and the function doesn't
+  ///                    fit in one block, thus requiring multiple batched transactions.
+  function withdrawRewards(
+    address staker,
+    uint104 batchSize
+  ) external override whenNotPaused returns (uint withdrawn, uint withdrawnUntilIndex) {
+    return _withdrawRewards(staker, staker, batchSize);
+  }
+
+  /// Withdraws a staker's accumulated rewards.
+  ///
+  /// @dev Only withdraws until the last finalized poll.
+  ///
+  /// @param destination The destination address where the rewards will be withdrawn.
+  /// @param batchSize   The index until which (but not including) the rewards should be withdrawn.
+  ///                    Used if a large number of assessments accumulates and the function doesn't
+  ///                    fit in one block, thus requiring multiple batched transactions.
+  function withdrawRewardsTo(
+    address destination,
+    uint104 batchSize
+  ) external override whenNotPaused returns (uint withdrawn, uint withdrawnUntilIndex) {
+    return _withdrawRewards(msg.sender, destination, batchSize);
+  }
+
+  function _withdrawRewards(
+    address staker,
+    address destination,
+    uint104 batchSize
+  ) internal returns (uint withdrawn, uint withdrawnUntilIndex) {
+    // This is the index until which (but not including) the previous withdrawal was processed.
+    // The current withdrawal starts from this index.
+    uint104 rewardsWithdrawableFromIndex = stakeOf[staker].rewardsWithdrawableFromIndex;
+    {
+      uint voteCount = votesOf[staker].length;
+      require(rewardsWithdrawableFromIndex < voteCount, "No withdrawable rewards");
+      // If batchSize is a non-zero value, it means the withdrawal is going to be batched in
+      // multiple transactions.
+      withdrawnUntilIndex = batchSize > 0 ? rewardsWithdrawableFromIndex + batchSize : voteCount;
+    }
+
+    Vote memory vote;
+    Assessment memory assessment;
+    for (uint i = rewardsWithdrawableFromIndex; i < withdrawnUntilIndex; i++) {
+      vote = votesOf[staker][i];
+      assessment = assessments[vote.assessmentId];
+      if (assessment.poll.end + config.payoutCooldownInDays * 1 days >= block.timestamp) {
+        // Poll is not final
+        withdrawnUntilIndex = i;
+        break;
+      }
+
+      withdrawn += uint(assessment.totalReward) * uint(vote.stakedAmount) /
+        (uint(assessment.poll.accepted) + uint(assessment.poll.denied));
+    }
+
+    // This is the index where the next withdrawReward call will start iterating from
+    stakeOf[staker].rewardsWithdrawableFromIndex = uint104(withdrawnUntilIndex);
+    ITokenController(getInternalContractAddress(ID.TC)).mint(destination, withdrawn);
```

Comment:
check the destination address is a member - the mint function (unfortunately) doesn't do that